include/boost/geometry/geometries/geometry_collection.hpp

```diff
@@ -0,0 +1,79 @@
+// Boost.Geometry
+
+// Copyright (c) 2021, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Adam Wulkiewicz, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_GEOMETRIES_GEOMETRY_COLLECTION_HPP
+#define BOOST_GEOMETRY_GEOMETRIES_GEOMETRY_COLLECTION_HPP
+
+#include <vector>
+
+#include <boost/geometry/core/tag.hpp>
+#include <boost/geometry/core/tags.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace model
+{
+
+/*!
+\brief Basic geometry_collection class representing a container of DynamicGeometries.
+\ingroup geometries
+\tparam DynamicGeometry Type adapted to DynamicGeometry Concept.
+\tparam Container \tparam_container
+\tparam Allocator \tparam_allocator
+*/
+template
+<
+    typename DynamicGeometry,
+    template <typename, typename> class Container = std::vector,
+    template <typename> class Allocator = std::allocator
+>
+class geometry_collection
+    : public Container<DynamicGeometry, Allocator<DynamicGeometry>>
+{
+    typedef Container<DynamicGeometry, Allocator<DynamicGeometry>> base_type;
+
+public:
+    geometry_collection() = default;
+
+//    template <typename Iterator>
+//    geometry_collection(Iterator begin, Iterator end)
+//        : base_type(begin, end)
+//    {}
```

Comment:
Delete this?

Reply:
Or add ctors in all models (linestring, ring, etc.) taking a pair of iterators?

Reply:
We might do that, but as far as I know, no user asked for this.

Reply:
No user asked for the `initializer_list` either. This is just something I found useful at some point during testing. But I'm ok with leaving it as it is.

Reply:
OK (we can always add it later if necessary)